### PR TITLE
Revise greater

### PR DESCRIPTION
**Pull request by bszmelcz**
### Details:
 - update spec
 - add template test
 - remove backend tests

### Tickets:
 - 37441

### CODE REVIEW

File: docs/ops/comparison/Greater_1.md [hunk 1]
```diff
@@ -4,7 +4,16 @@
 
 **Category**: Comparison binary operation
 
-**Short description**: *Greater* performs element-wise comparison operation with two given tensors applying multi-directional broadcast rules.
+**Short description**: *Greater* performs element-wise comparison operation with two given tensors applying broadcast rules specified in the `auto_broadcast` attribute.
+
+**Detailed description**
+Before performing arithmetic operation, input tensors *a* and *b* are broadcasted if their shapes are different and `auto_broadcast` attributes is not `none`. Broadcasting is performed according to `auto_broadcast` value.
+
+After broadcasting *Greater* does the following with the input tensors *a* and *b*:
```
Comment (tsavina) [inline docs/ops/comparison/Greater_1.md:12]:
```suggestion
After broadcasting, *Greater* does the following with the input tensors *a* and *b*:
```

File: docs/ops/comparison/Greater_1.md [hunk 2]
```diff
@@ -4,7 +4,16 @@
 
 **Category**: Comparison binary operation
 
-**Short description**: *Greater* performs element-wise comparison operation with two given tensors applying multi-directional broadcast rules.
+**Short description**: *Greater* performs element-wise comparison operation with two given tensors applying broadcast rules specified in the `auto_broadcast` attribute.
+
+**Detailed description**
+Before performing arithmetic operation, input tensors *a* and *b* are broadcasted if their shapes are different and `auto_broadcast` attributes is not `none`. Broadcasting is performed according to `auto_broadcast` value.
```
Comment (tsavina) [inline docs/ops/comparison/Greater_1.md:10]: `auto_broadcast` attributes is not ---> either `auto_broadcast` attribute is or `auto_broadcast` attributes are

**Review by tsavina — APPROVED** [review]

**Review by ggalieroc — APPROVED** [review]

File: docs/template_plugin/tests/functional/op_reference/greater.cpp [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    return compParams;
+}
+
+std::vector<RefComparisonParams> generateComparisonCombinedParams() {
+    const std::vector<std::vector<RefComparisonParams>> compTypeParams {
+        generateComparisonParams<element::Type_t::f32>(element::f32),
+        generateComparisonParams<element::Type_t::f16>(element::f16),
+        generateComparisonParams<element::Type_t::i32>(element::i32),
+        generateComparisonParams<element::Type_t::u32>(element::u32),
+        generateComparisonParams<element::Type_t::u8>(element::boolean)};
```
Comment (ilyachur) [inline docs/template_plugin/tests/functional/op_reference/greater.cpp:69]:
```suggestion
        generateComparisonParams<element::Type_t::boolean>(element::boolean)};
```
?

File: docs/template_plugin/tests/functional/op_reference/greater.cpp [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+            .input2({{1, 2, 1}, type, std::vector<T> {1, 1}})
+            .expected({{2, 1, 2, 1}, element::boolean, std::vector<char> {1, 0, 1, 0}})};
+    return compParams;
+}
+
+std::vector<RefComparisonParams> generateComparisonCombinedParams() {
+    const std::vector<std::vector<RefComparisonParams>> compTypeParams {
+        generateComparisonParams<element::Type_t::f32>(element::f32),
+        generateComparisonParams<element::Type_t::f16>(element::f16),
+        generateComparisonParams<element::Type_t::i32>(element::i32),
```
Comment (ilyachur) [inline docs/template_plugin/tests/functional/op_reference/greater.cpp:67]: Looks like i64 u64 data types are not covered

**Review by ilyachur — APPROVED** [review]

**Pull request merged**